### ISSUE TITLE
Firefox version?

### ISSUE THREAD

**Issue by Piter432** [Eloston/disable-html5-autoplay#40]
Like in title. Of course, if it's necessary

**Comment by Eloston** [comment]
Yes, it's still necessary. Unless the situation has changed, Firefox's `media.autoplay.enabled` will simply prevent untrusted `play()` invocations on all pages, which can send media players to bad states (they expect events to be fired or property values to change). Also, there's no way to control which pages will have autoplay disabled.

To quote the README:
> Firefox will be supported when [WebExtensions](https://wiki.mozilla.org/WebExtensions) has matured enough to support all the features necessary.

I took a quick look just now, and WebExtensions supports mostly everything that's needed for the extension to work. I'll hold off on this until I've finished version 0.8 first. By then, the extension will be able to stop autobuffering.

**Comment by xxcriticxx** [comment]
any eta on this?

**Comment by Eloston** [comment]
@xxcriticxx Not sure yet. Maybe after version 0.8.

**Comment by xxcriticxx** [comment]
and what version we in right now?

**Comment by Eloston** [comment]
@xxcriticxx The latest version can be found [on the releases page](https://github.com/Eloston/disable-html5-autoplay/releases). Planned changes for each new version are grouped under [milestones](https://github.com/Eloston/disable-html5-autoplay/milestones). Version 0.7 will most likely take more time to do than 0.8.

**Comment by xxcriticxx** [comment]
thx for info keep up great work

**Comment by uBlock-user** [comment]
Setting `media.playback.enabled` to `false` does the job, so why bother doing the same thing via an addon ?

**Comment by Piter432** [comment]
@uBlock-user: I know about that switch in `about:config`. Also, I know a lot about `about:config`. I mean I have heavily modified Firefox on my old PC (yes, I have lot of powerful extensions and Firefox doesn't slow on my old PC in opposite to Chrome with much lower number of Chrome's extensions, while lot of them isn't very powerful, if you know what I mean).

Btw., this switch disables auto play on **_ALL_** websites, if I need to enable auto play temporary I need to switch it to `true` and refresh the page if the setting didn't work without doing refreshing the page. The way that I can disable/enable auto play on the page in this extension is better. Not the best (there should be something like in dynamic panel in uBlock Origin for users who turn on advanced features of uBO), but better than setting this switch in `about:config` to `true`/`false`.

**Comment by xxcriticxx** [comment]
@Piter432 i think you can do this with uMatrix

**Comment by uBlock-user** [comment]
>Btw., this switch disables auto play on ALL websites, if I need to enable auto play temporary I need to switch it to true and refresh the page if the setting didn't work without doing refreshing the page.

Yes, In that case, you will need an extension. You didn't mention this in the original message or anything of that sort and no one said anything about this switch so I presumed none of you knew about it.

Anyways for me, that switch did the job, so I shared the info here. Also this extension is fabolous and the only extension I know that was able to stop BRIGHTCOVE players from making multiple attempts to invoke the autoplay on Comicbook.com. That website is riddled with these players on all articles.

**Comment by Eloston** [comment]
> Someone already made an extension for FF - https://addons.mozilla.org/En-us/firefox/addon/disable-autoplay/

I briefly looked at the code and it seems YouTube-specific right now. I don't see any code that is supposed to work for all websites.

Sure this extension doesn't quite work the way it should on many websites, but it works better than disabling JavaScript or HTML5 media completely, which is sure to break all websites. Eventually I do plan to add a more advanced language to control the extension's behavior much like uBlock Origin.

Unfortunately I don't have an ETA when any major development is going to happen again. I have many priorities outside of my two popular GitHub projects (this is one of them), and for the time being, I feel like diverting my limited time to ungoogled-chromium of the two projects. This doesn't mean I'm abandoning Disable HTML5 Autoplay though; that I am sure of.

**Comment by uBlock-user** [comment]
Hey great job so far, I'm loving the extension, these new players were making it worse for folks like me who are on limited data plans and hate the annoyance of a video starting on its own everytime you refresh a page or turn to another article. Adding the pre-load feature will be an absolute bonus though, so I hope you do that soon.

**Comment by GuardianMajor** [comment]
Confirmed on the Aurora branch - ie. `53.0a2 (2017-02-05) (64-bit)` the option is `media.autoplay.enabled` which needs to be set to **`false`** as stated by Eloston, so far as I can see the option provided by uBlock-user is either deprecated, obsolete or no longer used. For clarity people should check.

**Comment by ahmedtds** [comment]
Such a shame we need an extension for this and even need to wait more than 1 year to have it in FF. (No offense my frustration is for this awful auto-play trend and browsers which don't give you proper control over it)

**Comment by Eloston** [comment]
> The implementations are by no means consistent. That alone makes any cross browser single solution difficult if not impossible.

There is some discussion to fork and modify uBlock Origin as a basis for the 0.7 rewrite. That will solve most of the incompatibility issues I believe. If not, it will make porting to Firefox easier (as Chrome and Firefox have a very similar feature set most of the time now).

**Comment by markygnlg** [comment]
Just thought that I should add:
+1!
I'm a very old supporter of firefox but Opera has won me over for the past few years (currency conversion via text highlight, auto popup for ..., chrome/chromium base which allows extensions via Install Chrome Extensions)
 . 
{EDIT:} Just found the option for this in v102.0.1 (64-bit):
Menu -> Settings -> Privacy & Security -> Permissions -> Autoplay -> Settings -> Default For All Websites -> [XXXXX]
